File: core/src/de/project/ice/ecs/components/NameComponent.java
package de.project.ice.ecs.components;


public class NameComponent implements IceComponent<NameComponent>
{
    public String name = "";

    @Override
    public void reset()
    {
        name = "";
    }

    @Override
    public void copyTo(NameComponent copy)
    {
        copy.name = name;
    }
}

File: core/src/de/project/ice/ecs/components/NameComponent.kt
package de.project.ice.ecs.components

import de.project.ice.annotations.Property


class NameComponent : CopyableIceComponent {
    @Property("The name of this entity")
    var name = ""

    override fun reset() {
        name = ""
    }

    override fun copyTo(copy: CopyableIceComponent) {
        if (copy is NameComponent) {
            copy.name = name
        }
    }
}
